x-pack/plugins/infra/public/components/logging/log_flyout.tsx

```diff
@@ -14,24 +14,50 @@ import {
   EuiToolTip,
 } from '@elastic/eui';
 import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
+import moment from 'moment';
 import React from 'react';
-
 import euiStyled from '../../../../../common/eui_styled_components';
+import { TimeKey } from '../../../common/time';
 import { InfraLogItem, InfraLogItemField } from '../../graphql/types';
 import { InfraLoadingPanel } from '../loading';
+
 interface Props {
   flyoutItem: InfraLogItem | null;
-  hideFlyout: () => void;
+  setFlyoutVisibility: (visible: boolean) => void;
   setFilter: (filter: string) => void;
+  setTarget: (target: TimeKey) => void;
+  setSurroundingLogsId: (id: string) => void;
   intl: InjectedIntl;
   loading: boolean;
+  stopLiveStreaming: () => void;
 }
 
 export const LogFlyout = injectI18n(
-  ({ flyoutItem, loading, hideFlyout, setFilter, intl }: Props) => {
+  ({
+    flyoutItem,
+    loading,
+    setFlyoutVisibility,
+    setFilter,
+    setTarget,
+    setSurroundingLogsId,
+    intl,
+    stopLiveStreaming,
+  }: Props) => {
     const handleFilter = (field: InfraLogItemField) => () => {
       const filter = `${field.field}:"${field.value}"`;
       setFilter(filter);
+
+      if (flyoutItem && flyoutItem.timestamp && flyoutItem.tiebreaker) {
+        const timestampMoment = moment(flyoutItem.timestamp);
+        if (timestampMoment.isValid()) {
+          setTarget({
+            time: timestampMoment.valueOf(),
+            tiebreaker: flyoutItem.tiebreaker,
+          });
+          setSurroundingLogsId(flyoutItem.id);
+          stopLiveStreaming();
```

Comment:
I really like how you composed the `onChange` and `onSubmit` handlers passed to the autocompletion component on the page level to avoid introducing additional knowledge into it.

Could we pass in such an composite `jumpToTarget({ time, tiebreaker, id })` prop, which on the page level is composed to do the three things it does here too?

That way there would be a lot less coupling between the flyout and these features.